core/arch/arm/mm/tee_mm.c

```diff
@@ -190,13 +196,13 @@ tee_mm_entry_t *tee_mm_alloc(tee_mm_pool_t *pool, size_t size)
 
 			if (remaining < size)
 				/* out of memory */
-				return NULL;
+				goto out;
 		}
 	}
 
 	nn = tee_mm_add(entry);
```

Comment:
I don't think, that this is big deal. `malloc()` spinlock protects `malloc` internal state, `tee_mm_pool` spinlock protects pool internal state. Also, I don't see easy way to move allocation before spinlock tacking, because `tee_mm_add()` entry not only allocates memory, but also updates linked list.